src/bms/ast/ast_parse.rs

```diff
@@ -18,11 +20,23 @@ pub(super) fn parse_control_flow_ast<'a>(
             Unit::RandomBlock { value, if_blocks } => {
                 // Select branch
                 let branch_val = match value {
-                    BlockValue::Random { max } => {
-                        if max == BigUint::from(0u64) {
+                    BlockValue::Random { ref max } => {
+                        if *max == BigUint::from(0u64) {
                             BigUint::from(0u64)
                         } else {
-                            rng.generate(BigUint::from(1u64)..=max)
+                            let expected: RangeInclusive<BigUint> =
+                                BigUint::from(1u64)..=max.clone();
+                            let v = rng.generate(expected.clone());
+                            if v < BigUint::from(1u64) || v > *max {
```

Comment:
```suggestion
                            if expected.contains(&v) {
```
Please use [`contains` method](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html#method.contains) for check.
Line 80 is same.